### ISSUE TITLE
Dedicated package-lock files

### ISSUE THREAD

**Issue by Pezmc** [flowforge/flowforge-dev-env#20]
### Description

Right now, all FlowForge applications share the same package lock _if_ they are installed via https://github.com/flowforge/flowforge-dev-env.

However, on staging, production, and on some developers locals, [flowforge](https://github.com/flowforge/flowforge) and the sister packages, are installed without a lock file. This has the potential to introduce small hard to reproduce bugs at best, and break production or introduce security issues at worst.

Without a lock file, there's no guarantee what version of a package will be installed as part of `npm install`. The installer will resolve to the "best" version _at the time of install_. This means that local/staging/production can all be running different versions. Production could even run on a newer version of a package than is running on staging.

To fix this, FlowForge repos should introduce their own package-lock files. This gives an absolute guarantee of what version of a package will be installed, and ensures consistency.

In order to do that, we either have to stop using [npm workspace](https://docs.npmjs.com/cli/v9/using-npm/workspaces?v=true) or find a way to have our servers all use the [flowforge-dev-env](https://github.com/flowforge/flowforge-dev-env) to do our installs. I propose the former as a much easier route.

The main downsides, is that npm installs will take slightly longer, as the same version of a package might not be shared across all repos, and loss of npm workspace conveniences.

```[tasklist]
### Repos
- [ ] https://github.com/flowforge/forge-ui-components/pull/144
- [ ] https://github.com/flowforge/flowforge-driver-k8s/pull/89
- [ ] https://github.com/flowforge/flowforge-device-agent/pull/121
- [ ] https://github.com/flowforge/flowforge-nr-project-nodes/pull/33
- [ ] https://github.com/flowforge/flowforge-nr-file-nodes/pull/21
- [ ] https://github.com/flowforge/flowforge-driver-docker/pull/61
- [ ] https://github.com/flowforge/flowforge-driver-localfs/pull/82
- [ ] https://github.com/flowforge/flowforge/pull/2324
- [ ] https://github.com/flowforge/flowforge-file-server/pull/55
- [ ] https://github.com/flowforge/flowforge-nr-persistent-context/pull/23
```

**Comment by Pezmc** [comment]
@hardillb Is there anything I've overlooked as to our production deployment and build process that this might impact? As far as I can see, as long as the repo is pulled in, the standard `npm install` will work as it always has (though we should swap to use of `npm ci` anywhere automated).

**Comment by hardillb** [comment]
All the production (and staging) builds are done from clean git clones of the individual repos, not the flowforge-dev-env. So if each repo has it's own lock file then these should be used as part of any build.

And yes, currently the build scripts do all use `npm install`

**Comment by hardillb** [comment]
https://github.com/npm/rfcs/issues/554

**Comment by Pezmc** [comment]
For now, we will be:

 - Stopping use of npm workspaces, this is designed for mono-repos, our use doesn't match their expectation
 - Start making use of `npm link` automated via #21 

We'll make these changes just after the next release.

In the mean time, all the package-lock PR's can be merged. 

Care will need to be taken if updating package.json files if your local is still using npm workspaces. **You must update and commit the package-lock.json or CI will fail.**

**Comment by Pezmc** [comment]
Fixed in #22 and #21
